tests/benchmark/README.md

```diff
@@ -0,0 +1,320 @@
+* `DeleteLabel<algo_t>` - Delete one label per iteration (meaning 1 vector for single value index, and multiple vector for multi value index).  
+`algo_t` is a specific index type (including multi/single) because we call `GetDataByLabel()`, which is only known to HNSW_Single/Multi and BruteForce_Single/Multi. This why this test case is defined separately in each .cpp file using `DEFINE_DELETE_LABEL` macro (defined in bm_vecsim_basics.h). The registration is done like all other test cases in the initialization file.  
+* `Range_BF` 
+* `Range_HNSW`  
+In addition, in this file we define macros with the arguments for the `Range`, `AddLabel` and `DeleteLabel` registration.
+### Remarks
+The [basic benchmark set](#bm_vecsimbasics) includes functions defined both in `BM_VecSimCommon` and `BM_VecSimBasics`. Pay attention to refer the correct class when defining the test in the [initialization file](#how-to-define-and-register-a-new-test).
+### class BM_VecSimUpdatedIndex<index_type_t> : public BM_VecSimCommon<index_type_t>
+Defined in bm_updated_index.h  
+In this class constructor we add the additional indices, the "updated" indices. The "before" indices are initialized in `BM_VecSimIndex`.  
+This class defines no new test routines and uses only tests defined in `BM_VecSimCommon`.  
+### Remarks
+The constructor is called after we already defined the tests resides in `BM_VecSimCommon`, so `ref_count` is not zero and we cant rely on it to decide weather we should initialize the indices or not. This is why we have to hold another the `is_initialized` flag.  
+Also, we keep the value of `ref_count` at the moment of initialization in `first_updatedBM_ref_count` to free the indices when `ref_count` decreased to this value.
+### class BM_BatchIterator<index_type_t> : public BM_VecSimIndex<index_type_t>
+Defined in bm_batch_iterator.h  
+Here we implement the tests routines of [batch iterator set](#bm_batchiterator) to be called by the test definition in the [initialization file](#how-to-define-and-register-a-new-test).  
+In this file we also define the arguments for each test case in the batch iterator tests set.
+
+### Tests counters
+* Tests counters finale values are printed in the benchmarks output next to the test results. 
+* They can also be used as input data in the Grafana dashboard.
+### Tests naming:
+A test name will look something like this:  
+`test_fixture<fixture_data_type>/test_name/arg1:val/iterations:val`  
+for example:  
+`BM_VecSimBasics<fp32_index_t>/Range_HNSW_Single/radiusX100:20/epsilonX1000:1/iterations:100`  
+So test's name components are:
+1. The test's fixture and the fixture type (if it's templated)
+2. The name as given at definition and registration
+3. arguments names and values  
+*NOTE:* the fixture type (fp32_index_t, for example) is part of the name, so tests that use different fixture **or** the same fixture of a different type, **can** share the same name.  
+
+
+# Cmake remarks
```

Comment:
you can remove this

Reply:
this = the note + cmake remarks section?